File: JavaScript_Mitre10Cup/JavaScript_Mitre10Cup/src/Game.js
/* globals View */
class Game {  // eslint-disable-line no-unused-vars
  constructor (newRound, theHomeTeam, theAwayTeam, newWhen) {
    this.round = newRound
    this.when = newWhen
    this.myHomeTeam = theHomeTeam
    // added code
    this.homeTeamScore = 0
    this.awayTeamScore = 0
    this.homeTeamTries = 0
    this.awayTeamTries = 0

    this.myAwayTeam = theAwayTeam
    if (theHomeTeam) {
      this.venue = theHomeTeam.getVenue()
    } else {
      this.venue = ''
    }
  }

  setResult (newTeamAScore, newTeamBScore) {
    this.homeTeamScore = newTeamAScore
    this.awayTeamScore = newTeamBScore
  }

  getDate (aDate) {
    var days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday']
    let day = days[aDate.getDay()]
    let date = aDate.getDate()
    let result = `${day} ${date}`
    return result
  }
  getTime (aDate) {
    let h = aDate.getHours()
    let m = aDate.getMinutes()
    let result = `${h}.${m}pm`
    return result
  }
  getMonth (aDate) {
    var months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
    return months[aDate.getMonth()]
  }
  getSunday () {
    let daysToSunday = 6 - this.when.getDay()
    let sunday = new Date(this.when)
    sunday.setDate(this.when.getDate() + daysToSunday)
    return `${this.getDate(sunday)} ${this.getMonth(sunday)}`
  }
  getWhen () {
    return `${this.getDate(this.when)} ${this.getMonth(this.when)}`
  }
  toString () {
    let result = this.getDate(this.when)
    let paddingNeeded = 13 - result.length
    let padding = View.SPACES(paddingNeeded)
    result += padding
    result += `${this.getTime(this.when)}${View.SPACES(2)}${this.myHomeTeam.name} v ${this.myAwayTeam.name} at ${this.venue}`
    return result
  }
  get () {
    let result = this.getDate(this.when)
    let paddingNeeded = 13 - result.length
    let padding = View.SPACES(paddingNeeded)
    result += padding
    result += `${this.getTime(this.when)} ${this.getMonth(this.when)}${View.NEWLINE()}${View.TAB()}${this.myHomeTeam.name} v ${this.myAwayTeam.name} at ${this.venue}`
    return result
  }
  // added code
  getResult () {
    let result = this.getDate(this.when)
    let paddingNeeded = 13 - result.length
    let padding = View.SPACES(paddingNeeded)
    result += padding
    // console.log("Whats my score")
    // console.log(this.homeTeamScore)
    result += `Result is ${this.myHomeTeam.name} ${this.homeTeamScore} : ${this.awayTeamScore} ${this.myAwayTeam.name}`
    // console.log(result)
    return result
  }

  addingScores (aGame, homeTeamScore, awayTeamScore, homeTeamTries, awayTeamTries) {
    this.setResult(homeTeamScore, awayTeamScore)
    aGame.myHomeTeam.for += homeTeamScore
    aGame.myHomeTeam.against += awayTeamScore
    if (homeTeamTries >= 4) {
      aGame.myHomeTeam.BP1 += 1
    }

    if (awayTeamScore - homeTeamScore <= 7 && awayTeamScore - homeTeamScore >= 1 ) {
      aGame.myHomeTeam.BP2 += 1
    }

    aGame.myAwayTeam.for += awayTeamScore
    aGame.myAwayTeam.against += homeTeamScore
    if (awayTeamTries >= 4) {
      aGame.myAwayTeam.BP1 += 1
    }

    if (homeTeamScore - awayTeamScore <= 7 && homeTeamScore - awayTeamScore >= 1 ) {
      aGame.myAwayTeam.BP2 += 1
    }
  }
  // --------------------------------------------------------------------------
  hasTeam (targetTeamName) {
    return this.myHomeTeam.name === targetTeamName || this.myAwayTeam.name === targetTeamName
  }
  isCrossOver () {
    let homeDivision = Math.trunc((this.myHomeTeam.rank - 1) / 7)
    let awayDivision = Math.trunc((this.myAwayTeam.rank - 1) / 7)
    let isDifferentDivision = homeDivision !== awayDivision
    return isDifferentDivision
  }
}

File: python_mitre10_cup/python_mitre10_cup/game.py
class Game(object):
    def __init__(self, new_week, the_home_team, the_away_team, new_date):
        self.week = new_week
        self.when = new_date
        self.home_team = the_home_team
        self.away_team = the_away_team
        self.homeTeamScore = 0
        self.awayTeamScore = 0
        if the_home_team:
            self.venue = the_home_team.get_venue()

        else:  # is this needed if NOT adding the semis and finals?
            self.venue = ''

    def __str__(self):
        date = self.when.strftime('%A %d')
        date = date.ljust(15)
        date += self.when.strftime('%I.%M%p')
        result = f'\t{date}\t{self.home_team} v {self.away_team}'
        result = result.ljust(60)
        result += self.venue
        return result

    def get(self):
        date = self.when.strftime('%A %d %B')
        date = date.ljust(25)
        date += self.when.strftime('%I.%M%p')
        result = f'\t{date}\t{self.home_team} v {self.away_team}'
        result = result.ljust(70)
        result += self.venue
        return result

    def has_team(self, target_team_name):
        return self.home_team.name == target_team_name or self.away_team.name == target_team_name

    def is_crossover(self):
        home_division = (self.home_team.rank - 1) // 7
        away_division = (self.away_team.rank - 1 ) // 7
        is_different_division = home_division != away_division
        return is_different_division

    def set_result(self, newTeamAScore, newTeamBScore):
        self.homeTeamScore = newTeamAScore
        self.awayTeamScore = newTeamBScore

#    code added by Tom Son
    def adding_scores (self, aGame, homeTeamScore, awayTeamScore, homeTeamTries, awayTeamTries):
        self.set_result(homeTeamScore, awayTeamScore)
        aGame.home_team.scored += homeTeamScore
        aGame.home_team.against += awayTeamScore
        if (homeTeamTries >= 4):
            aGame.home_team.BP1 += 1

        if (awayTeamScore - homeTeamScore <= 7 and awayTeamScore - homeTeamScore >= 1):
            aGame.home_team.BP2 += 1

        aGame.away_team.scored += awayTeamScore
        aGame.away_team.against += homeTeamScore
        if (awayTeamTries >= 4):
            aGame.away_team.BP1 += 1

        if (homeTeamScore - awayTeamScore <= 7 and homeTeamScore - awayTeamScore >= 1):
            aGame.away_team.BP2 += 1


